galsim/chromatic.py

```diff
@@ -1567,9 +1661,26 @@ def drawImage(self, bandpass, image=None, integrator='trapezoidal', **kwargs):
             jac, offset, _ = self._getTransformations(bandpass.red_limit)
             int_im = galsim.Transform(int_im, jac=jac, offset=offset, gsparams=self.gsparams)
             image = int_im.drawImage(image=image, **kwargs)
+            self._last_wcs = image.wcs
             return image
         else:
-            return ChromaticObject.drawImage(self, bandpass, image, integrator, **kwargs)
+            image = ChromaticObject.drawImage(self, bandpass, image, integrator, **kwargs)
+            self._last_wcs = image.wcs
+            return image
+
+    @property
+    def noise(self):
```

Comment:
>I think we should go through and move all the noise propagation to noise property functions

Sounds good to me.  Looking into it...

>GSObject and Chromatic object can have default implementations that return None,

I think this may be problematic, since `GSObject.__init__` can have another `GSObject` passed in, and we want to propagate the noise of that object.  So I think we might need to have the base object noise still be a regular attribute so it can be overwritten and vary instance to instance.  We can certainly make `Transform.noise` a property though (and `Sum`, `Convolve`, ...), and delay noise computations for them.